cloud-provider-snapshot-restore-jobs/cmd/resource/resource.go

```diff
@@ -0,0 +1,216 @@
+package resource
+
+import (
+	"context"
+	"fmt"
+	"github.com/aws-cloudformation/cloudformation-cli-go-plugin/cfn/encoding"
+	"github.com/aws-cloudformation/cloudformation-cli-go-plugin/cfn/handler"
+	matlasClient "github.com/mongodb/go-client-mongodb-atlas/mongodbatlas"
+	"github.com/mongodb/mongodbatlas-cloudformation-resources/util"
+)
+
+// Create handles the Create event from the Cloudformation service.
+func Create(req handler.Request, prevModel *Model, currentModel *Model) (handler.ProgressEvent, error) {
+	client, err := util.CreateMongoDBClient(*currentModel.ApiKeys.PublicKey.Value(), *currentModel.ApiKeys.PrivateKey.Value())
+	if err != nil {
+		return handler.ProgressEvent{}, err
+	}
+
+	deliveryType := currentModel.DeliveryType.Value()
+	if deliveryType == nil || (*deliveryType != "automated" && *deliveryType != "download") {
+		return handler.ProgressEvent{
+			OperationStatus: handler.Failed,
+			Message:         "error creating cloud provider snapshot restore job: you need to implement only one, `automated` and `download` delivery types",
+			ResourceModel:   currentModel,
+		}, nil
+	}
+	targetClusterName := currentModel.TargetClusterName.Value()
+	targetProjectId := currentModel.TargetProjectId.Value()
+	if *deliveryType == "automated" {
+		if targetClusterName == nil || *targetClusterName == "" {
+			return handler.ProgressEvent{
+				OperationStatus: handler.Failed,
+				Message:         "error creating cloud provider snapshot restore job: `target_cluster_name` must be set if delivery type is `automated`",
+				ResourceModel:   currentModel,
+			}, nil
+		}
+		if targetProjectId == nil || *targetProjectId == "" {
+			return handler.ProgressEvent{
+				OperationStatus: handler.Failed,
+				Message:         "error creating cloud provider snapshot restore job: `target_project_id` must be set if delivery type is `automated`",
+				ResourceModel:   currentModel,
+			}, nil
+		}
+	}
+	requestParameters := &matlasClient.SnapshotReqPathParameters{
+		GroupID:     *currentModel.ProjectId.Value(),
+		ClusterName: *currentModel.ClusterName.Value(),
+	}
+	snapshotRequest := &matlasClient.CloudProviderSnapshotRestoreJob{
+		SnapshotID:   *currentModel.SnapshotId.Value(),
+		DeliveryType: *deliveryType,
+	}
+	if *deliveryType == "automated" {
```

Comment:
if you move tove Line 45 and 46 before the same comparation against automated you can do this inside the same if

Reply:
Also, you might want to use a constant.